### PR TITLE
[TASK] Introduce Makefile and command 'make docs'

### PR DESCRIPTION
**Pull request by linawolf**
This is more convenient then having to copy-paste the command all the time

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+.PHONY: help
+help: ## Displays this list of targets with descriptions
+	@echo "You prepend/append the argument 'ENV=(local|docker)' to each target. This specifies,"
+	@echo "whether to execute the target within your local environment, or docker (default).\n"
```
Comment (garvinhicking) [inline Makefile:4]: This doesn't apply to this makefile, the env params are only evaluated in render-guides, the docker container doesn't utilize it.

**Review by garvinhicking — CHANGES_REQUESTED** [review]

**Comment by garvinhicking** [comment]
Generally I like the idea to add that to each repository that uses the new rendering. One could also add a global command individually, so any project could be rendered, but the convenience in this one as a consistent command is nice!

File: Makefile (hunk 1 above)
Comment (linawolf) [reply Makefile]: Ack

**Review by garvinhicking — APPROVED** [review]

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -5,6 +5,15 @@ Contributing
 Information About Contributing to This Manual
 =============================================
 
+Local rendering
+---------------
+
+You can render this file locally if you have `Docker <https://www.docker.com/>`
```
Comment (brotkrueml) [inline CONTRIBUTING.rst:11]: Which file? The contributing?

File: CONTRIBUTING.rst [hunk 3]
```diff
@@ -5,6 +5,15 @@ Contributing
...
+and `make <https://www.gnu.org/software/make/>` installed on your local
+machine::
+
+    make docs
```
Comment (brotkrueml) [inline CONTRIBUTING.rst:15]: And what is the alternative for WIndows users?

**Review by brotkrueml — COMMENTED** [review]
I hope, we don't want to add a Makefile to all manuals?

File: CONTRIBUTING.rst (hunk 3 above)
Comment (linawolf) [reply CONTRIBUTING.rst:15]: WSL2.

**Comment by linawolf** [comment]
> I hope, we don't want to add a Makefile to all manuals?

That would be my plan for the official manuals. And I would like to supply a snippet so ppl can add it to their extensions

File: CONTRIBUTING.rst (hunk 2 above)
Comment (linawolf) [inline CONTRIBUTING.rst:11]:
```suggestion
You can render this manual locally if you have `Docker <https://www.docker.com/>`
```

Comment (linawolf) [reply CONTRIBUTING.rst]: changed it to manual

File: CONTRIBUTING.rst (hunk 3 above)
Comment (brotkrueml) [reply CONTRIBUTING.rst:15]: So we force the users to use Linux/Mac.

What about providing a `render.sh` and a `render.bat` file?

**Comment by brotkrueml** [comment]
> > I hope, we don't want to add a Makefile to all manuals?
> 
> That would be my plan for the official manuals. And I would like to supply a snippet so ppl can add it to their extensions

Then we definitly need to provide a solution for Windows users.

**Comment by linawolf** [comment]
I am using windows, this runs fine on WSL2 and is preinstalled. I don't know of any windows user that develops TYPO3 but does not use WSL2. And this script is not mandatory, just a convenient shortcut

**Comment by brotkrueml** [comment]
Okay, then go ahead, you have already an approval.

**Comment by brotkrueml** [comment]
And: Then we should document the way that works every time.

**Comment by brotkrueml** [comment]
And to copy/paste: I am using Ctrl + R for searching in the zsh/bash history. It is faster than typing render.sh manually ;-)

**Comment by linawolf** [comment]
For those who prefere the Docker command it still works as before...

File: CONTRIBUTING.rst (hunk 3 above)
Comment (linawolf) [reply CONTRIBUTING.rst:15]: We force no-one. they can use the Docker command as before?

**Comment by brotkrueml** [comment]
... which is not documented here.

**Comment by garvinhicking** [comment]
We could theoretically add a composer command that executes docker for the official doc repositories. That would then not have a Make/wsl2/cygwin dependency.

We could also have both of course, but I feel render-guides overdid this already.

Providing a rendering shortcut that is
- used in all official repos
- easy to run
- hides potential complexity in those long "docker run" commands
- documented
- not a requirement
- not hindering alternate rendering ways (local, manual)
- easy to maintain with low overhead

is an improvement to me? I was more used to composer scripts until I joined the project, and I was under the impression "make" was a decission made already.

**Comment by brotkrueml** [comment]
IMHO Composer scripts should not be used for calling some binary stuff unrelated to PHP itself. There are other ways to do so. If the command is the same for Windows/Linux/Macos, then put it into a file called "render-docs" and that's it.

Just my 2 cents.

**Comment by garvinhicking** [comment]
We'd have to add render-docs.sh and render-docs.bat then, I think that would be even less elegant than a Makefile or composer script. And clutter some directory if we add more calls to be executed (guides.xml conversion, guides.xml modification, xml linting...). I think we need a central place for it, so either Makefile (as now) or composer. Everything else would have more dependencies (ie "just" or other command runners).

**Pull request merged**